client/repositories/signature/signature.go

```diff
@@ -0,0 +1,105 @@
+package signature_repo
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	"github.com/lidofinance/dc4bc/client/types"
+	"github.com/syndtr/goleveldb/leveldb"
+)
+
+const (
+	SignaturesKeyPrefix = "signatures"
+)
+
+type state interface {
```

Comment:
is it a kind of interface segregation principle implementation? if yes, in my opinion, it's a bad one:
1. if the original State interface changes (client/modules/state/state.go) in one of these methods, you will have to change this little copy of it as well, so your decoupling doesn't work;
2. you don't use all the defined state methods in the code below, only Get and Set.

Moreover, according to the [refactoring roadmap](https://www.notion.so/DC4BC-Refactoring-Roadmap-7eb5c992aae94c828d2dc6700693eb7a), we will have the State interface split to three different ones, and the Offset- and Operations-related methods will be torn off from the Get, Set and Delete. So I'd simply use the State interface in the SignatureRepo:

```golang
type SignatureRepo struct {
	stateDb state.State
}
```

please correct me if I'm wrong